lib/api/multiObjectDelete.js

```diff
@@ -118,58 +195,74 @@ function _parseXml(xmlToParse, next) {
 * @callback called with (err, quietSetting, errorResults, numOfObjects,
 * successfullyDeleted, totalContentLengthDeleted)
 */
-export function getObjMetadataAndDelete(bucketName, quietSetting,
-    errorResults, inPlay, log, next) {
+export function getObjMetadataAndDelete(authInfo, canonicalID, request,
+        bucketName, bucket, quietSetting, errorResults, inPlay, log, next) {
     const successfullyDeleted = [];
     let totalContentLengthDeleted = 0;
     let numOfObjects = 0;
-    // for obj deletes, no need to check acl's at object level
-    // (authority is at the bucket level for obj deletes)
 
     // doing 5 requests at a time. note that the data wrapper
     // will do 5 parallel requests to data backend to delete parts
-    return async.forEachLimit(inPlay, 5, (key, moveOn) => {
-        metadata.getObjectMD(bucketName, key, {}, log, (err, objMD) => {
-            // if general error from metadata return error
-            if (err && !err.NoSuchKey) {
-                log.error('error getting object MD', { error: err, key });
-                errorResults.push({
-                    key,
-                    error: err,
-                });
-                return moveOn();
-            }
-            // if particular key does not exist, AWS returns success
-            // for key so add to successfullyDeleted list and move on
-            if (err && err.NoSuchKey) {
-                successfullyDeleted.push(key);
-                return moveOn();
-            }
-            return services.deleteObject(bucketName, objMD, key, log,
-                err => {
-                    if (err) {
-                        log.error('error deleting object', { error: err, key });
-                        errorResults.push({
-                            key,
-                            error: err,
-                        });
+    return async.forEachLimit(inPlay, 5, (entry, moveOn) => {
+        async.waterfall([
+            // for obj deletes, no need to check acl's at object level
+            // (authority is at the bucket level for obj deletes)
+            callback => metadataGetObject(bucketName, entry.key,
+                entry.versionId, log, (err, objMD) => {
+                    // if general error from metadata return error
+                    if (err && !err.NoSuchKey) {
+                        log.error('error getting object MD',
+                                { error: err, key: entry.key });
+                        errorResults.push({ entry, error: err });
                         return moveOn();
```

Comment:
This should be callback as well.
Edit: This can fixed in another PR since this commit is not the one introducing it.